widgetssdk/src/main/java/com/glia/widgets/call/CallController.java

```diff
@@ -231,7 +231,9 @@ public void onHoldChanged(boolean isOnHold) {
     public void engagementEnded() {
         Logger.d(TAG, "engagementEndedByOperator");
         stop();
-        dialogController.dismissDialogs();
+        if (!Glia.getCurrentEngagement().isPresent()) {
```

Comment:
But what if present? The call screen will be stuck?
Maybe better fix the GliaEngagementStateRepository instead of fixing behavior on its events.

Reply:
There is a task for investigating anomaly behaviours with engagement states. This is to fix acceptance tests now